dts/arm/atmel/sam4e.dtsi

```diff
@@ -21,6 +21,14 @@
 			device_type = "cpu";
 			compatible = "arm,cortex-m4f";
 			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
```

Comment:
This entry is not really needed - Cortex-M4 has a fixed number of 8 regions.

Reply:
Hi @ioannisg, I prefer keep the entries, it could be important on future Zephyr versions. On my perspective it should be dropped only when it wont be necessary for all SoCs.

Reply:
I agree with @nandojve here. Most (if not all) of the MPU-equipped SoCs seem to have the `mpu` entry and we should add one here for the sake of consistency; it is generally a good practice to retrieve hardware parameters from the device tree too, instead of hard-coding in the code.

Reply:
No @stephanosio that cannot be accurate. AFAIK, only Cortex-M7 and Cortex-M33 SoCs have the mpu node defined.

Reply:
@ioannisg You are right; that does seem to be the case:
https://github.com/zephyrproject-rtos/zephyr/search?p=1&q=extension%3Adtsi+e000ed90&unscoped_q=extension%3Adtsi+e000ed90

I wonder if we should add all the missing `mpu` entries for the all the MPU-equipped SoCs. cc @galak

Reply:
We can - I was thinking this could eventually lead to having MPU enabled entirely.
But it is not required for now and most M3,M4 SoCs don't do that.

Reply:
well, as long as we set `status = "disabled";` for the MPU, the MPU should be considered to be disabled (and handled as such in the code).

sure, adding MPU entries for the rest would be beyond the scope of this PR. maybe we should remove these entries until we do a refactoring in the future -- no strong opinion on this one.

Reply:
agreed

Reply:
I think @galak already started changes on #24386. This way, should we wait?

Reply:
no, I think #24386 does not add mpu nodes in dts for existing SoCs - this work is orthogonal to that one.

Reply:
But we could wait, since #24386 would potentially remove the fixup entries for mpu and nvic